src/DIRAC/Core/Tornado/Server/BaseRequestHandler.py

```diff
@@ -0,0 +1,739 @@
+      bound to disappear.
+  """
+  # Because we initialize at first request, we use a flag to know if it's already done
+  __init_done = False
+  # Lock to make sure that two threads are not initializing at the same time
+  __init_lock = threading.RLock()
+
+  # MonitoringClient, we don't use gMonitor which is not thread-safe
+  # We also need to add specific attributes for each service
+  _monitor = None
+
+  # System name with which this component is associated
+  SYSTEM = None
+
+  # Auth requirements
+  AUTH_PROPS = None
+
+  # Type of component
+  MONITORING_COMPONENT = MonitoringClient.COMPONENT_WEB
+
+  # Prefix of methods names
+  METHOD_PREFIX = "export_"
+
+  # Which grant type to use
+  USE_AUTHZ_GRANTS = ['SSL', 'JWT']
+
+  # Definition of identity providers
+  _idps = IdProviderFactory()
+  _idp = {}
+
+  @classmethod
+  def _initMonitoring(cls, serviceName, fullUrl):
+    """
+      Initialize the monitoring specific to this handler
+      This has to be called only by :py:meth:`.__initializeService`
+      to ensure thread safety and unicity of the call.
+
+      :param serviceName: relative URL ``/<System>/<Component>``
+      :param fullUrl: full URl like ``https://<host>:<port>/<System>/<Component>``
+    """
+
+    # Init extra bits of monitoring
+
+    cls._monitor = MonitoringClient()
+    cls._monitor.setComponentType(cls.MONITORING_COMPONENT)
+
+    cls._monitor.initialize()
+
+    if tornado.process.task_id() is None:  # Single process mode
+      cls._monitor.setComponentName('Tornado/%s' % serviceName)
+    else:
+      cls._monitor.setComponentName('Tornado/CPU%d/%s' % (tornado.process.task_id(), serviceName))
+
+    cls._monitor.setComponentLocation(fullUrl)
+
+    cls._monitor.registerActivity("Queries", "Queries served", "Framework", "queries", MonitoringClient.OP_RATE)
+
+    cls._monitor.setComponentExtraParam('DIRACVersion', DIRAC.version)
+    cls._monitor.setComponentExtraParam('platform', DIRAC.getPlatform())
+    cls._monitor.setComponentExtraParam('startTime', datetime.utcnow())
+
+    cls._stats = {'requests': 0, 'monitorLastStatsUpdate': time.time()}
+
+    return S_OK()
+
+  @classmethod
+  def _getServiceName(cls, request):
+    """ Search service name in request.
+
+        :param object request: tornado Request
+
+        :return: str
+    """
+    raise NotImplementedError('Please, create the _getServiceName class method')
+
+  @classmethod
+  def _getServiceAuthSection(cls, serviceName):
+    """ Search service auth section.
+
+        :param str serviceName: service name
+
+        :return: str
+    """
+    raise NotImplementedError('Please, create the _getServiceAuthSection class method')
+
+  @classmethod
+  def _getServiceInfo(cls, serviceName, request):
+    """ Fill service information.
+
+        :param str serviceName: service name
+        :param object request: tornado Request
+
+        :return: dict
+    """
+    return {}
```

Comment:
Shouldn't there be a warning of "non-implementation"?

Reply:
IMHO: this method is optional for child classes. Warning message added